app/models/exercise_with_solution.rb

```diff
@@ -0,0 +1,25 @@
+class ExerciseWithSolution < SimpleDelegator
```

Comment:
This is fine for this PR, but I wonder if we should create a new directory to move this into and keep `models` as `AR Models`. What do you think?

Reply:
I don't have any preference. Some teams do with a separate `decorators/` and some don't. Let's see what works for us. I'll keep it in here for the meantime.